src/commands/base-command.mjs

```diff
@@ -456,6 +459,21 @@ export default class BaseCommand extends Command {
 
     const globalConfig = await getGlobalConfig()
 
+    // Get framework, add to analytics payload for every command, if a framework is set
+    const fs = new NodeFS()
+    const project = new Project(fs, buildDir)
+    const frameworks = await project.detectFrameworks()
+
+    const frameworkIDs = frameworks?.map((framework) => framework.id)
+
+    if (frameworkIDs?.length !== 0) {
+      this.setAnalyticsPayload({ frameworks: frameworkIDs })
+    }
+
+    const buildSystem = await project.detectBuildSystem()
+    const packageManager = await project.detectPackageManager()
```

Comment:
That's not needed as `detectFrameworks()` depends on it and run those first, so it just retrieves the value from the `project`.

you can do directly `{ packageManager: project.packageManager?.name, buildSystem: project.buildSystems.map(({id}) => id) }`

Reply:
![CleanShot 2023-05-26 at 08 53 46](https://github.com/netlify/cli/assets/11156362/a36f9537-dd9e-4eb2-a003-b86b2b43f14c)